### DISCUSSION TITLE
Reactive Key-Value keys

### DISCUSSION THREAD

**Discussion by lochinvarwest** [filamentphp/filament#12477] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.71

### How can we help you?

I have a Key-Value field on a form.  When creating the record, the keys must not be editable, but the available keys must be based on the value of another field - e.g. if the category is "Books", I want the keys in the Key-Value field to be ["Author", " Value"] but if the category field is "Clothes" the available keys must be ["Colour", "Value"].  BTW - The category field already has the ->live() function...
I can't find a way to reactively define the available keys.  Any help?

**Comment by leandrocfe** [comment, marked as answer]
```php
Select::make('category')
    ->options([
        'books' => 'Books',
        'clothes' => 'Clothes',
    ])
    ->live()
    ->afterStateUpdated(function (Select $component, ?string $state) {

        $keyValue = $component->getContainer()->getComponent('dynamic_attributes');

        if ($state) {
            $keyValue->state(match ($state) {
                'books' => ['author' => 'John Doe', 'title' => 'xyz'],
                'clothes' => ['color' => 'Red'],
                default => [],
            });
        }

    }),
Forms\Components\KeyValue::make('attributes')
    ->addable(false)
    ->deletable(false)
    ->editableKeys(false)
    ->key('dynamic_attributes')
```

**Reply by lochinvarwest** [reply]
Awesome - rather than listening from the KeyValue, 'push' it from the Category - with the small change that if the KeyValue is invisible, the container can't find the 'dynamic_attributes' component - so I just left it visible but without any keys. Thank you!

As an aside - The Select field has '->options' in which you can use a Get to determine the new value of a different live field - seeing as they are both filled by an array, for consistency I would suggest the KeyValue field also has ->options().

**Reply by leandrocfe** [reply]
Nice! I updated the code as your suggestion ;)
